core/src/test/java/org/apache/struts2/interceptor/parameter/StrutsParameterAnnotationTest.java

```diff
@@ -343,6 +360,14 @@ public Map<String, Pojo> getPublicPojoMapDepthTwo() {
         }
     }
 
-    class Pojo {
+    static class ModelAction implements ModelDriven<Pojo> {
```

Comment:
There is one more case where `ModelDriven<Object>` is used and then `getModel()` can return conditional model, either a `List<Pojo>` or `Pojo` itself. See [that example](https://github.com/apache/struts-examples/blob/master/rest-angular/src/main/java/org/apache/examples/struts/actions/data/OrderController.java#L71-L80)

Reply:
Ah interesting, let me test this

Reply:
I think in such cases, the app developer should manually OGNL allowlist any required types, I'm not confident of a secure way to auto allowlist in this case

Reply:
Yeah, it can be tricky, but having such a note should be enough 👍